runners/flink/runner/src/main/java/org/apache/beam/runners/flink/FlinkRunner.java

```diff
@@ -176,4 +259,368 @@ public String toString() {
     }
     return files;
   }
+
+  /**
+   * Specialized implementation for
+   * {@link org.apache.beam.sdk.transforms.View.AsMap View.AsMap}
+   * for the Flink runner in streaming mode.
+   */
+  private static class StreamingViewAsMap<K, V>
+      extends PTransform<PCollection<KV<K, V>>, PCollectionView<Map<K, V>>> {
+
+    @SuppressWarnings("unused") // used via reflection in FlinkRunner#apply()
+    public StreamingViewAsMap(View.AsMap<K, V> transform) {
+    }
+
+    @Override
+    public PCollectionView<Map<K, V>> apply(PCollection<KV<K, V>> input) {
+      PCollectionView<Map<K, V>> view =
+          PCollectionViews.mapView(
+              input.getPipeline(),
+              input.getWindowingStrategy(),
+              input.getCoder());
+
+      @SuppressWarnings({"rawtypes", "unchecked"})
+      KvCoder<K, V> inputCoder = (KvCoder) input.getCoder();
+      try {
+        inputCoder.getKeyCoder().verifyDeterministic();
+      } catch (Coder.NonDeterministicException e) {
+//        runner.recordViewUsesNonDeterministicKeyCoder(this);
```

Comment:
If I understand the flow of code & data correctly for this and the multimap case, the `KV<K, V>` are concatenated into a list which is then converted by `MapPCollectionView` (via `MapViewFn`) into a conventional Java `Map<K, V>` using Java-level equality.

So for now I don't think a deterministic coder is necessary for your runner. This has to change for scalable cross-language map side inputs, but it seem OK for now. So instead of calling `verifyDeterministic` and throwing away the exception, these 5 lines can just be wiped out (and below).